fire-hose-management-frontend/src/components/Departments/EditDepartmentOverlay.jsx

```diff
@@ -0,0 +1,186 @@
+import {makeStyles} from "@material-ui/core/styles";
+import React, {useState} from "react";
+import {Button, Checkbox, FormControlLabel, Grid, MenuItem} from "@material-ui/core";
+import Stack from "@mui/material/Stack";
+import {Select, TextField} from "@mui/material";
+import {CSelect} from "../shared/input/CSelect";
+import {Overlay} from "../shared/Overlay";
+import {ValidatorForm} from "react-material-ui-form-validator";
+import {putDepartment} from "../../utils/requests/Departments";
+import {useMutation} from "react-query";
+import Notifications from "../shared/Notifications";
+import {CTextField} from "../shared/input/CTextField";
+
+const useStyles = makeStyles((theme) => ({
+    checkbox: {
+        display: "table-cell",
+        verticalAlign: "middle"
+    },
+    footer: {
+        paddingTop: "24px",
+        paddingBottom: "32px"
+    }
+}));
+
+const genderMenuItems = [<MenuItem key={0} value={"MALE"}>Herr</MenuItem>,
+    <MenuItem key={1} value={"FEMALE"}>Frau</MenuItem>,
+    <MenuItem key={2} value={"OTHER"}>Divers</MenuItem>]
+
+const EditDepartmentOverlay = ({
+                                   department,
+                                   handleClose
+                               }) => {
+
+    const classes = useStyles();
+
+    const [gender, setGender] = useState(department.contact.gender)
+    const [firstname, setFirstname] = useState(department.contact.firstname)
+    const [lastname, setLastname] = useState(department.contact.lastname)
+    const [mail, setMail] = useState(department.contact.mail)
+    const [sendConfirmationMail, setSendConformationMail] = useState(department.sendConfirmationMail)
+
+    const editDepartmentMutation = useMutation((departmentDto) => putDepartment(department.id, departmentDto), {
+        onSuccess: () => {
+            Notifications.showSuccess("Die Abteilung wurde erfolgreich aktualisiert!")
+
+        },
+        onError: (error) => {
+            Notifications.showError("Die Abteilung konnte nicht aktualisiert werden!")
+            console.log(error)
+        }
+    })
+
+
+    const departmentBuilder = (gender, firstname, lastname, mail, sendConfirmationMail) => {
+        return {
+            gender: gender,
+            firstname: firstname,
+            lastname: lastname,
+            mail: mail,
+            sendConfirmationMail: sendConfirmationMail
+        }
+    }
+
+
+    console.log(department)
+
+    return (<Overlay
+        onClose={() => handleClose()}
+        headerContent={<h2>{department.departmentName} aktualisieren</h2>}
+        size={"m"}>
+        <ValidatorForm
+            onSubmit={() => editDepartmentMutation.mutate(departmentBuilder(gender, firstname, lastname, mail, sendConfirmationMail))}>
```

Comment:
Codacy has a fix for the issue: [Replace `····onSubmit={()·=>·editDepartmentMutation.mutate(departmentBuilder(gender,·firstname,·lastname,·mail,·sendConfirmationMail))}>` with `onSubmit={()·=>⏎··········editDepartmentMutation.mutate(⏎············departmentBuilder(⏎··············gender,⏎··············firstname,⏎··············lastname,⏎··············mail,⏎··············sendConfirmationMail⏎············)⏎··········)⏎········}`](https://app.codacy.com/gh/StitZle/fire-hose-complete/pullRequest?prid=9542260)
```suggestion
        onSubmit={() =>
```